deepreg/model/backbone/global_net.py

```diff
@@ -81,18 +79,16 @@ def call(self, inputs, training=None, mask=None):
         :return:
         """
         # down sample from level 0 to E
-        h_in = inputs
         for level in range(self._extract_max_level):  # level 0 to E - 1
-            h_in, _ = self._downsample_blocks[level](inputs=h_in, training=training)
+            h_in, _ = self._downsample_blocks[level](inputs=inputs, training=training)
         h_out = self._conv3d_block(
             inputs=h_in, training=training
         )  # level E of encoding
 
         # predict affine parameters theta of shape = [batch, 4, 3]
-        theta = self._dense_layer(h_out)
-        theta = self._reshape(theta)
-
+        self.theta = self._dense_layer(h_out)
+        self.theta = self._reshape(self.theta)

```

Comment:
for this `_reshape`, maybe we can remove it from the init, it's a layer without variables and we can do `tf.reshape(self.theta, shape=(-1,4,3))` (a function) directly on it?

Reply:
Yes we can